ngi_pipeline/engines/piper_ngi/launchers.py

```diff
@@ -46,165 +48,190 @@
+                                              generate_bqsr_bam=analysis_object.generate_bqsr_bam)
+                    if analysis_object.exec_mode == "sbatch":
+                        process_id = None
+                        slurm_job_id = sbatch_piper_sample([setup_xml_cl, piper_cl],
+                                                           workflow_subtask,
+                                                           analysis_object.project, sample,
+                                                           restart_finished_jobs=analysis_object.restart_finished_jobs,
+                                                           files_to_copy=default_files_to_copy)
+                        for x in xrange(10):
+                            # Time delay to let sbatch get its act together
+                            # (takes a few seconds to be visible with sacct)
+                            try:
+                                get_slurm_job_status(slurm_job_id)
+                                break
+                            except ValueError:
+                                time.sleep(2)
+                        else:
+                            LOG.error('sbatch file for sample {}/{} did not '
+                                      'queue properly! Job ID {} cannot be '
+                                      'found.'.format(analysis_object.project, sample, slurm_job_id))
+                    else: # "local"
+                        raise NotImplementedError('Local execution not currently implemented. '
+                                                  'I\'m sure Denis can help you with this.')
+                        #slurm_job_id = None
+                        #launch_piper_job(setup_xml_cl, project)
+                        #process_handle = launch_piper_job(piper_cl, project)
+                        #process_id = process_handle.pid
+                    try:
+                        record_process_sample(project=analysis_object.project,
+                                              sample=sample,
+                                              analysis_module_name="piper_ngi",
+                                              slurm_job_id=slurm_job_id,
+                                              process_id=process_id,
+                                              workflow_subtask=workflow_subtask)
+                    except RuntimeError as e:
+                        LOG.error(e)
+                        ## Question: should we just kill the run in this case or let it go?
+                        continue
+                except (NotImplementedError, RuntimeError, ValueError) as e:
+                    error_msg = ('Processing project "{}" / sample "{}" / workflow "{}" '
+                                 'failed: {}'.format(analysis_object.project, sample,
+                                                     workflow_subtask,
+                                                     e))
+                    LOG.error(error_msg)
```

Comment:
The main issue that caused me to work on this is that the current code assumes that we want to start analysis on sample level, no matter the engine. Unfortunately, this doesnt hold true for the next two engines, namely RNA and Cancer. as such, I needed to move big chunks of code around, and the obnoxious amount of parameters that have to be carried over different functions is really something I'm not a fan of. 
FV is still on holidays until the end of this week, and I'll leave afterwards, so we might put this on hold.